boxsdk/pagination/page.py

```diff
@@ -61,7 +61,7 @@ def __getitem__(self, key):
         item_class = self._translator.translate(item_json['type'])
         item = item_class(
             session=self._session,
-            object_id=item_json['id'],
+            object_id=item_json['id'] if 'id' in item_json else None,
```

Comment:
Note that `RecentItem` is not a subclass of `BaseObject` because most of the functionality provided by `BaseObject` does not apply to to recent items (o way to GET individual recent items, no way to update). Finally, recent items from the API do not have unique IDs associated with them, so there's no way to get `item_json['id']`